internal/host/hostcommands/install_cmd.go

```diff
@@ -321,7 +321,10 @@ func constructHostInstallerArgs(cluster *common.Cluster, host *models.Host, inve
 
 	// Check if Manufacturer is IBM/S390 and ProductName is not "KVM/Linux" (the case for z/VM and LPAR).
 	// If this is the case than we need to extract the necessary z/VM kargs and append them.
-	if inventory.SystemVendor != nil && strings.EqualFold(inventory.SystemVendor.Manufacturer, ZVM_VENDOR_ID) && !strings.HasSuffix(inventory.SystemVendor.ProductName, VM_CTRL_PRG) {
+	log.Debugf("Check for SystemVendor and ProductName: %s:%s\n", inventory.SystemVendor.Manufacturer, inventory.SystemVendor.ProductName)
+	if inventory.SystemVendor != nil && strings.EqualFold(inventory.SystemVendor.Manufacturer, ZVM_VENDOR_ID) &&
```

Comment:
Maybe this can be shorter ?

```
if inventory.SystemVendor != nil && strings.EqualFold(inventory.SystemVendor.Manufacturer, ZVM_VENDOR_ID) &&
(inventory.SystemVendor.ProductName == "" || !strings.HasSuffix(inventory.SystemVendor.ProductName, VM_CTRL_PRG)) {
```

Reply:
Thx, good catch. Fixed